bcgov/HOUS-permit-portal
# Bugfix/bphh 1057/add specs and delete pipeline

## Description

Refactored the code for a more succinct upload process.  No syncing of submission_data payload with supporting documents to reduce edge cases.
More consistent removal of item (even if save is not triggered or svae is triggered).
Updated the parsing of signatures information.

## What type of PR is this? (check all applicable)


## Related Tickets & Documents

(https://hous-bssb.atlassian.net/browse/BPHH-992)
(https://hous-bssb.atlassian.net/browse/BPHH-1057)

## Steps to QA
file uploads should still work.
computed compliance should still work.
After uploading, and compliance is run, if you remove the document the compliance will clear.
If you delete persisted file and do not save, you can refresh the screen and download it.
